documentation/intermediate_ca_configuration.markdown

```diff
@@ -0,0 +1,286 @@
+
+```
+curl -k https://$(puppet agent --configprint server):8140/puppet-ca/v1/certificate/ca > $(puppet agent --configprint localcacert)
+```
+
+### Puppet agent has `certificate_revocation` set to true
+
+#### Error message
+
+```
+Warning: SSL_connect returned=1 errno=0 state=error: certificate verify failed: [unable to get certificate CRL for /CN=Puppet Enterprise CA generated on <CA server fqdn> at +2017-10-03 00:13:16 +0000]
+```
+
+This error message looks similar to the error produced by a CRL with the wrong issuer. The difference between the two cases is that if `certificate_revocation` is incorrectly set, the subject/CN will be the subject of the CA certificate, while a CRL with the wrong issuer will generate an error mentioning the hostname of the Puppet Master.
+
+#### Cause
+
+The default value for `certificate_revocation` is true, but running Puppet with this default setting with an intermediate CA causes SSL verification to fail.
+
+#### Solution
+
+Configure the Puppet agent to perform only leaf certificate revocation checking by setting `certificate_revocation` to `leaf`.
+
+```
+puppet config set --section main certificate_revocation leaf
+```
+
+As an insecure alternative, you can configure the Puppet agent to disable certificate revocation checking by setting `certificate_revocation` to `false`. However, this prevents Puppet Server from successfully revoking certificates, which can open your infrastructure to attacks from agents that have revoked credentials.
+
+```
+puppet config set --section main certificate_revocation false
+```
+
+### Puppet agent has a CRL with the wrong issuer
+
+#### Error message
+
+```
+Warning: SSL_connect returned=1 errno=0 state=error: certificate verify failed: [unable to get certificate CRL for /CN=<Puppet Server fqdn]
+```
+
+This error message looks similar to the error produced by an incorrect `certificate_revocation` setting. The difference between the two cases is that if `certificate_revocation` is incorrectly set, the subject/CN will be the subject of the CA certificate, while a CRL with the wrong issuer will generate an error mentioning the hostname of the Puppet Master.
+
+#### Cause
+
+Puppet does not verify that a CRL was issued by a specific CA, so it's possible to distribute and update a CRL whose issuer is not the Puppet CA. In addition, the `puppet cert revoke` and `puppet cert clean` commands will modify a CRL issued by a different CA and save the new CRL with the wrong issuer.
+
+Because the issuer of the CRL does not match the issuer of the Puppet Server host, Puppet is unable to determine that the loaded CRL is associated with the certificate. This mismatch can be verified by comparing the issuer of the host certificate and the downloaded CRL.
+
+```
+diff -q <(openssl x509 -in $(puppet agent --configprint hostcert) -noout -issuer) <(openssl crl -in $(puppet agent --configprint hostcrl) -noout -issuer)
```

Comment:
windows version

Reply:
Replicating this functionality on Windows doesn't seem trivial - @Iristyle do you know how to do something like this on Windows?